src/hooks/useFetch.jsx

```diff
@@ -12,19 +14,27 @@ import { useEffect, useState } from "react";
  * useFetch
  * Custom hook for fetching data from the API
  * @param {useFetchOptions} options
- * @returns {object} { data, fetching, error, fetchData, resetState }
+ * @returns {object} { data, pagination, fetching, error, fetchData, resetState }
  */
 const useFetch = (options) => {
   const [data, setData] = useState(null);
+  const [pagination, setPagination] = useState(null);
   const [fetching, setFetching] = useState(false);
   const [error, setError] = useState(null);
   const [abortController, setAbortController] = useState(null);
-  const { dataSubSelector, dataSetter, fetchingSetter, errorSetter } =
-    options || {};
+  const {
+    dataSubSelector,
+    dataSetter,
+    paginationSelector,
+    paginationSetter,
+    fetchingSetter,
+    errorSetter,
```

Comment:
I'm not entirely sure why these setters need to be passed in when they are also created above via `useState`:
```js
const [pagination, setPagination] = useState(null);
```

Reply:
I'm not sure what the original reason was for implementing `useFetch` this way, but it was interfering with testing, so I made `useFetch` retrieve state directly from the `SearchContext`.